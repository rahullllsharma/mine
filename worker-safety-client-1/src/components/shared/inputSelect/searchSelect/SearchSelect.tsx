import type { InputSelectOption, InputSelectProps } from "../InputSelect";
import InputSelect from "../InputSelect";

export type SearchSelectProps = Omit<
  InputSelectProps<InputSelectOption>,
  "isSearchable"
>;

export default function SearchSelect({
  onSelect,
  ...props
}: SearchSelectProps): JSX.Element {
  const handleItemSelect = (
    item: InputSelectOption | readonly InputSelectOption[]
  ): void => onSelect(item as InputSelectOption);

  return <InputSelect {...props} isSearchable onSelect={handleItemSelect} />;
}
