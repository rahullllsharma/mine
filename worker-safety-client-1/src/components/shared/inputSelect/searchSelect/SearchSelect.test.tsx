import type { InputSelectOption } from "../InputSelect";
import { fireEvent, render, screen } from "@testing-library/react";
import SearchSelect from "./SearchSelect";

const DUMMY_TASKS: InputSelectOption[] = [
  { id: "task_1", name: "Cable tray and support install" },
  { id: "task_2", name: "Clearing and grading" },
  { id: "task_3", name: "Control line installation" },
  { id: "task_4", name: "Excavation of soil using hydro-vac" },
  { id: "task_5", name: "Clearing" },
];

describe("SearchSelect", () => {
  const selectHandler = jest.fn();

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchSelect
        options={DUMMY_TASKS}
        onSelect={selectHandler}
        className="w-72"
        placeholder="select options"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders valid options when user enters a value in the search input", () => {
    render(
      <SearchSelect
        options={DUMMY_TASKS}
        onSelect={selectHandler}
        className="w-72"
        placeholder="select options"
      />
    );
    const inputElement = screen.getByRole("combobox");
    const searchTerm = DUMMY_TASKS[4].name;

    fireEvent.change(inputElement, { target: { value: searchTerm } });

    const optionsDisplayed = screen.getAllByRole("option").length;
    const numberOfMatches = screen.getAllByRole("option", {
      name: new RegExp(searchTerm, "i"),
      exact: false,
    }).length;

    expect(optionsDisplayed).toEqual(numberOfMatches);
  });

  it("shouldn't render any options when the search term doesn't match any of the list items", () => {
    render(
      <SearchSelect
        options={DUMMY_TASKS}
        onSelect={selectHandler}
        className="w-72"
        placeholder="select options"
      />
    );
    const inputElement = screen.getByRole("combobox");
    const searchTerm = "random searchTerm";

    fireEvent.change(inputElement, { target: { value: searchTerm } });

    expect(screen.queryAllByRole("option")).toHaveLength(0);
  });
});
