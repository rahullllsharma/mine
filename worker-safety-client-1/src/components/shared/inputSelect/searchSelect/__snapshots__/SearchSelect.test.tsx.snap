// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSelect renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: relative;
  box-sizing: border-box;
}

.emotion-1 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 0 0 0;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
  gap: 4px;
}

.emotion-5 {
  color: hsl(0, 0%, 50%);
  grid-area: 1/1/2/3;
  margin-left: 2px;
  margin-right: 2px;
  box-sizing: border-box;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-6 {
  margin: 0;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: visible;
  color: hsl(0, 0%, 20%);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  box-sizing: border-box;
}

.emotion-6:after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0;
  outline: 0;
  padding: 0;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-8 {
  color: hsl(0, 0%, 80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-8:hover {
  color: hsl(0, 0%, 60%);
}

<div
    class="w-72 emotion-0"
  >
    <span
      class="emotion-1"
      id="react-select-2-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="emotion-1"
    />
    <div
      class="w-full pr-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 cursor-pointer border-neutral-shade-26 text-neutral-shade-58 pl-3 emotion-3"
      role="button"
    >
      <div
        class=" emotion-4"
      >
        <div
          class=" emotion-5"
          id="react-select-2-placeholder"
        >
          select options
        </div>
        <div
          class=" emotion-6"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-select-2-listbox"
            aria-describedby="react-select-2-placeholder"
            aria-expanded="false"
            aria-haspopup="true"
            aria-owns="react-select-2-listbox"
            autocapitalize="none"
            autocomplete="off"
            autocorrect="off"
            class=""
            id="react-select-2-input"
            role="combobox"
            spellcheck="false"
            style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
            tabindex="0"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class=" emotion-7"
      >
        <div
          aria-hidden="true"
          class=" emotion-8"
        >
          <i
            aria-hidden="true"
            class="ci-chevron_down text-neutral-shade-58 text-xl"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
